### ISSUE TITLE
Crash because of `DiaSourceAlt`

### ISSUE THREAD

**Issue by mrexodia** [horsicq/PDBRipper#8]
You use:

```
    HRESULT hr=NoRegCoCreate(L"msdia140.dll", _uuidof(DiaSourceAlt),
                              _uuidof(IDiaDataSource),
                              (void **)(&g_pDiaDataSource));
```

But this crashes (in Debug mode) because `DiaSourceAlt` uses a different string allocation strategy and `SysFreeString` will cause heap corruption. If you want to use `DiaSourceAlt` you have to use `LocalFree(bstr - 2)` (completely undocumented except for a wrong hint at https://www.developerfusion.com/article/84368/debugging-with-the-dia-sdk/)

In the past I did some benchmarks and using `DiaSourceAlt` isn't faster for loading symbols (source lines or globals) but I don't know about types. How would you prefer to handle this? I can switch to `DiaSource` or change all `SysFreeString` to `LocalFree`.

**Comment by horsicq** [comment]
Thanks a lot for the tip! I think we should just change it to DiaSource.

**Comment by mrexodia** [comment]
I’m now checking if there is any difference in the output to be sure

**Comment by horsicq** [comment]
Nice!

**Comment by mrexodia** [comment]
Closed by #9
